gtk/docs.md

```diff
@@ -1 +1,357 @@
 <!-- file * -->
+<!-- struct TextBuffer -->
+You may wish to begin by reading the
+[text widget conceptual overview](https://developer.gnome.org/gtk3/stable/TextWidget.html)
+which gives an overview of all the objects and data
+types related to the text widget and how they work together.
+<!-- impl TextBuffer::fn insert_markup -->
+Inserts the text in `markup` at position `iter`. `markup` will be inserted
+in its entirety and must be nul-terminated and valid UTF-8. Emits the
```

Comment:
We can get to finer details now. What do you think about cleaning up things irrelevant in Rust bindings? For instance `&str` always is utf-8 and we take care of nul-termination behind the scenes so any mentions of these are unnecessary and confusing.